Simplenote/src/main/java/com/automattic/simplenote/PreferencesFragment.java

```diff
@@ -456,6 +481,101 @@ public int compare(String text1, String text2) {
         }
     }
 
+    private void importData(Uri uri) {
+        try {
+            String exportContents = readFile(uri);
+            switch (getFileExtension(uri)) {
+                case "json":
+                    importJsonExport(exportContents);
+                    break;
+                case "txt":
+                    importPlaintextFile(exportContents);
+                    break;
+                case "md":
+                    importPlaintextFile(exportContents);
+                    break;
+                default:
+                    Toast.makeText(requireContext(), getString(R.string.import_unknown), Toast.LENGTH_SHORT).show();
+                    return;
+            }
+            Toast.makeText(requireContext(), getString(R.string.import_message_success), Toast.LENGTH_SHORT).show();
+        } catch (Exception e) {
+            Toast.makeText(requireContext(), getString(R.string.import_message_failure), Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    private void importPlaintextFile(String exportContents) {
+        Simplenote currentApp = (Simplenote) requireActivity().getApplication();
+        Bucket<Note> noteBucket = currentApp.getNotesBucket();
+        Note note = noteBucket.newObject();
+        note.setContent(exportContents);
+        note.setCreationDate(Calendar.getInstance());
+        note.setModificationDate(note.getCreationDate());
+        note.save();
+    }
+
+    private void importJsonExport(String exportContents) throws Exception {
+        JSONObject jsonData = new JSONObject(exportContents);
+        JSONArray activeNotes = jsonData.getJSONArray("activeNotes");
+        JSONArray trashedNotes = jsonData.getJSONArray("trashedNotes");
+        List<String> trashedIds = new ArrayList<>();
+        JSONArray notesArray = new JSONArray();
+        for (int i = 0; i < activeNotes.length(); i++) {
+            notesArray.put(activeNotes.getJSONObject(i));
+        }
+        for (int j = 0; j < trashedNotes.length(); j++) {
+            notesArray.put(trashedNotes.getJSONObject(j));
+            trashedIds.add(trashedNotes.getJSONObject(j).getString("id"));
+        }
+        for (int k = 0; k < notesArray.length(); k++) {
+            JSONObject noteJson = notesArray.getJSONObject(k);
+            noteFromJson(noteJson, trashedIds);
```

Comment:
it would be preferable if this function returned a note (for testing and independence etc…). if we made that change this line would become

```java
noteFromJson(noteJson, trashedIds).save();
```

Reply:
Okay, I'll do it now